app/src/sharedTest/java/org/oppia/android/app/classroom/ClassroomListFragmentTest.kt

```diff
@@ -0,0 +1,878 @@
+
+    val args = TopicActivityParams.newBuilder().apply {
+      this.topicId = FRACTIONS_TOPIC_ID
+      this.storyId = FRACTIONS_STORY_ID_0
+    }.build()
+    intended(hasComponent(TopicActivity::class.java.name))
+    intended(hasProtoExtra(TopicActivity.TOPIC_ACTIVITY_PARAMS_KEY, args))
+  }
+
+  @Test
+  fun testFragment_scrollToBottom_classroomListSticks_classroomListIsVisible() {
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).performScrollToIndex(3)
+    composeRule.onNodeWithTag(CLASSROOM_LIST_TEST_TAG).assertIsDisplayed()
+  }
+
+  @Test
+  fun testFragment_switchClassroom_topicListUpdatesCorrectly() {
+    // Click on Science classroom card.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_TEST_TAG).onChildAt(0).performClick()
+    testCoroutineDispatchers.runCurrent()
+
+    // Check that Science classroom's topics are displayed.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
+      .assertTextContains("First Test Topic")
+      .assertTextContains("3 Lessons")
+      .assertIsDisplayed()
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
+      .assertTextContains("Second Test Topic")
+      .assertTextContains("1 Lesson")
+      .assertIsDisplayed()
+
+    // Click on Maths classroom card.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_TEST_TAG).onChildAt(1).performClick()
+    testCoroutineDispatchers.runCurrent()
+
+    // Check that Maths classroom's topics are displayed.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
+      .assertTextContains("Fractions")
+      .assertTextContains("2 Lessons")
+      .assertIsDisplayed()
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
+      .assertTextContains("Ratios and Proportional Reasoning")
+      .assertTextContains("4 Lessons")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  fun testFragment_clickOnTopicCard_returnBack_classroomSelectionIsRetained() {
+    // Click on Maths classroom card.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_TEST_TAG).onChildAt(1).performClick()
+    testCoroutineDispatchers.runCurrent()
+
+    // Check that Fractions topic is displayed and perform click.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
+      .assertTextContains("Fractions")
+      .assertTextContains("2 Lessons")
+      .assertIsDisplayed()
+      .performClick()
+
+    pressBack()
+
+    // Check that Maths classroom is selected & its topics are displayed.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
+      .assertTextContains("Fractions")
+      .assertTextContains("2 Lessons")
+      .assertIsDisplayed()
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
+      .assertTextContains("Ratios and Proportional Reasoning")
+      .assertTextContains("4 Lessons")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  @Ignore("Temporarily ignored as the test is failing.")
```

Comment:
Making a note

Reply:
One more thing about this test setup,
  `testCoroutineDispatchers.registerIdlingResource()` needs to come before` profileTestHelper.initializeProfiles()` during initialization.

Reply:
I misunderstood the comment initally, updated the test per latest suggestion.